### ISSUE TITLE
Deletion of secrets in SecretsManager throws ResourceNotFoundException

### ISSUE THREAD

**Issue by gnashuk** [localstack/localstack#3798]
# Type of request: This is a ...

[x ] bug report
[ ] feature request

# Detailed description
There seems to be a problem with deleting created secrets using SecretsManager AWS service. The method deleteSecret() throws and exception `Caused by: software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException: Secrets Manager can't find the specified secret. (Service: SecretsManager, Status Code: 404, Request ID: null, Extended Request ID: null)`. But when checking the secrets in the service it properly shows that the specified secret was marked as deleted with deletion date not null. 
So, for some reason, the secret does get deleted by the operation but throws ResourceNotFoundException instead of success response. Issue observed on testcontainers versions 1.5.0 and 1.5.2.
...

## Expected behavior
deleteSecret() should return successful response
...

## Actual behavior
Throws `software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException`
...

# Steps to reproduce
1. Create `LocalStackContainer` with `SECRETSMANAGER` service
2. Create a secret in SecretsManager
3. Call `SecretsManagerClient.deleteSecret()` passing the created secret ARN
## Command used to start LocalStack

...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```
CreateSecretRequest createSecretRequest = createSecretRequestBuilder
                .name("test-secret")
                .secretString("test-value)
                .build();

CreateSecretResponse createSecretResponse = secretsManagerClient.createSecret(createSecretRequest);
String secretArn = createSecretResponse.arn();

DeleteSecretRequest deleteSecretRequest = DeleteSecretRequest.builder()
                .secretId(secretArn)
                .build();
secretsManagerClient.deleteSecret(deleteSecretRequest)
```
...

**Comment by mgagliardo** [comment]
Hello @gnashuk, I am not sure what would be the full code you run, but I have tested this out locally and it seems to be working with the latest Java SDK.

Code:
```
package com.localstack;

import java.net.URI;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
import software.amazon.awssdk.services.secretsmanager.model.CreateSecretRequest;
import software.amazon.awssdk.services.secretsmanager.model.DeleteSecretRequest;
import software.amazon.awssdk.services.secretsmanager.model.CreateSecretResponse;
import software.amazon.awssdk.services.secretsmanager.model.SecretsManagerException;


public class CreateSecret {

    public static void main(String[] args) throws Exception {

       final String USAGE = "\n" +
                "Usage:\n" +
                "    CreateSecret  <secretName> <secretValue> \n\n" +
                "Where:\n" +
                "    secretName - the name of the secret (for example, tutorials/MyFirstSecret). \n"+
                "    secretValue - the secret value. \n";

        String secretName = "test";
        String secretValue= "test";

        Region region = Region.US_EAST_1;
        SecretsManagerClient secretsClient = SecretsManagerClient.builder()
                .endpointOverride(URI.create("http://localhost:4566"))
                .region(region)
                .build();

       String secretArn = createNewSecret(secretsClient, secretName, secretValue);
       System.out.println("The secret ARN is "+ secretArn);
       System.out.println("Now Deleting Secret "+ secretArn);
       deleteSecret(secretsClient, secretArn);
       secretsClient.close();
    }

    public static String createNewSecret( SecretsManagerClient secretsClient, String secretName, String secretValue) {

        try {
            CreateSecretRequest secretRequest = CreateSecretRequest.builder()
                .name(secretName)
                .description("This secret was created by the AWS Secret Manager Java API")
                .secretString(secretValue)
                .build();

            CreateSecretResponse secretResponse = secretsClient.createSecret(secretRequest);
            return secretResponse.arn();

        } catch (SecretsManagerException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
        return "";
    }

    public static void deleteSecret( SecretsManagerClient secretsClient, String secretArn ) {
        DeleteSecretRequest deleteSecretRequest = DeleteSecretRequest.builder()
                        .secretId(secretArn)
                        .build();
        secretsClient.deleteSecret(deleteSecretRequest);
    }
}
```

Output (You can see the `DeletedDate` value):
```
$ awslocal secretsmanager list-secrets
{
    "SecretList": [
        {
            "ARN": "arn:aws:secretsmanager:us-east-1:000000000000:secret:test-xZEgx",
            "Name": "test",
            "Description": "This secret was created by the AWS Secret Manager Java API",
            "KmsKeyId": "",
            "RotationEnabled": false,
            "RotationLambdaARN": "",
            "RotationRules": {
                "AutomaticallyAfterDays": 0
            },
            "DeletedDate": 1619644065.023051,
            "Tags": [],
            "SecretVersionsToStages": {
                "9be878c1-d0b9-4a0d-a13d-0b135730cb5e": [
                    "AWSCURRENT"
                ]
            }
        }
    ]
}
```

Let me know if you run into any issues.

**Comment by gnashuk** [comment]
Hi @mgagliardo, thanks for taking a look at this. I was initialising a localstack with the Docker image version `localstack/localstack:0.12.6`. Now, I have tried the newer versions and the issue is not appearing there. Looks like this bug has been fixed in the version 0.12.7 - https://github.com/localstack/localstack/releases/tag/v0.12.7.

**Comment by mgagliardo** [comment]
Thanks. Closing this now.
